### ISSUE TITLE
Suport for TypeScript version selection with typesVersions

### ISSUE THREAD

**Issue by k3nsei** [angular/angular-cli#12486]
### Bug Report or Feature Request (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [x] feature request
```

### Command (mark with an `x`)
```
- [ ] new
- [x] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Versions

All

### Repro steps

N/A

### The log given by the failure

N/A

### Desired functionality

It would be nice to have build compatible with new TypeScript 3.1 feature [typesVersions](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-1.html#version-selection-with-typesversions). Tu support older version of angular when we building libraries.

**Comment by alan-agius4** [comment]
`typesVersions` doesn’t have any effect on versions of Angular used or Angular metadata. But rather the version of typings resolves.

The `typeVersions` should already work if you add this feature to your library and put the typings generates by an older version of TypeScript in a separate folder.

That said, we always aim to support the latest versions of upstream dependencies over older once.

**Comment by k3nsei** [comment]
Let me rephrase. I'm requesting for mechanism to generate or transform those typings for older versions of TypeScript. Like you can specify target EcmaScript version in tsconfig.

**Comment by clydin** [comment]
That seems to be a feature request for typescript and not the Angular CLI.
